### ISSUE TITLE
xbox series x controller showing dpad buttons as hat inputs

### ISSUE THREAD

**Issue by patrickelectric** [libsdl-org/SDL#7976]
I'm testing the joystick with sdl2 and it appears that besides the normal 16 buttons, sdl2 also reports that the xbox controller also has a hat, where the dpads shows as hat inputs.

That only happens under Windows, linux and mac are fine.

**Comment by slouken** [comment]
Yeah, there is no guarantee in the joystick API about what hats, axes, and buttons correspond to what parts of the joystick. In particular there are several controllers that report both axes and buttons for triggers, and PS3 controllers can report axes for all pressure sensitive buttons. If you want to treat a controller as a gamepad, you should use the SDL gamepad API.
